src/components/fields/secretField.tsx

```diff
@@ -61,15 +67,13 @@ export function SecretField({
 
   const handelCancelReset = () => {
     setFieldResetted(false);
-    setValue(name as any, secretFieldValue);
+    setValue(name as any, getDefaultValue(defaultValues, name));
   };
 
-  const isSecret = editMode && getDefaultValue(defaultValues, name) === secretFieldValue;
-  const isObscured = isSecret && !fieldResetted && getValues(name as any) === secretFieldValue;
-  const displayValue = isObscured
-    ? envPlaceholder
-    : // ? value.replace(/\$\{[^}]+\}/g, envPlaceholder.slice(0, 6)) TODO: partial obfuscation
-      value;
+  const isSecret = editMode && getDefaultValue(defaultValues, name).includes(secretFieldValue);
+  const isObscured =
+    isSecret && !fieldResetted && getValues(name as any).includes(secretFieldValue);
+  const displayValue = isObscured ? obfuscate(value) : value;
```

Comment:
Why `includes`?

Reply:
Adds support for the use case where only a substring is the secretFieldValue, like in:
<img width="760" alt="Screenshot 2024-09-05 at 3 25 33 PM" src="https://github.com/user-attachments/assets/99775c12-b988-4dcf-bb99-2ba40d4de695">